src/monodroid/jni/monodroid-glue-external.hh

```diff
@@ -0,0 +1,67 @@
+// Dear Emacs, this is a -*- C++ -*- header
+#ifndef __MONODROID_GLUE_EXTERNAL_HH
+#define __MONODROID_GLUE_EXTERNAL_HH
+
+#include <stdio.h>
+#include <stdarg.h>
+#include <jni.h>
+
+#include "monodroid.h"
+#include "logger.hh"
+#include "timing.hh"
+#include "xamarin_getifaddrs.h"
+
+namespace xamarin::android
+{
+	class MonodroidExternalAPI
```

Comment:
What I find _fascinating_ about this is that it *feels like* a reversion of 6be4bdbcfd9f2fbe6267aceb934394bcdd0f13b4, which is "Stop using dlopen/dlsym to load Mono symbols."  It's *not* -- there are far fewer `dlsym()` calls with this approach -- but it *feels* like it, and I just find that...funny.

Reply:
Unfortunately, it means that instead of a direct call to certain functions, it's a *virtual* call to certain functions.  *Some* overhead, but not much.

Reply:
There are no virtual calls involved, not really. Note that this class is pure virtual, it's an interface class, the one we derive from it, `MonodroidExternalAPI_Impl` is marked as `final` (like C# `sealed` really( and all of its methods are also `final`. The compiler will know it and will generate plain calls.